File: iOS链式编程/CalculatorForOC/CalculatorForOC/XZCalculator.m
//
//  XZCalculator.m
//  CalculatorForOC
//
//  Created by aa on 2017/11/7.
//  Copyright © 2017年 aa. All rights reserved.
//

#import "XZCalculator.h"

@implementation XZCalculator

- (XZCalculator *(^)(NSInteger num))add{
    return ^(NSInteger num){
        _result += num;
        NSLog(@"result %d",_result);
        return self;
    };
}

- (XZCalculator *(^)(NSInteger num))minus{
    return ^(NSInteger num){
        _result -= num;
        NSLog(@"result %d",_result);
        return self;
    };
}


- (XZCalculator *(^)(NSInteger num))multiply{
    return ^(NSInteger num){
        _result *= num;
        NSLog(@"result %d",_result);
        return self;
    };
}


- (XZCalculator *(^)(NSInteger num))divide{
    return ^(NSInteger num){
        _result /= num;
        NSLog(@"result %d",_result);
        return self;
    };
}

@end

File: iOS链式编程/CalculatorForSwift/CalculatorForSwift/XZCalculator.swift
//
//  XZCalculator.swift
//  CalculatorForSwift
//
//  Created by aa on 2017/11/7.
//  Copyright © 2017年 aa. All rights reserved.
//

import UIKit

class XZCalculator: NSObject {
    
    var result:Int = 0;

    func add(_ num:Int) -> XZCalculator {
        self.result += num;
        print("result \(self.result)");
        return self;
    }
    
    func minus(_ num:Int) -> XZCalculator {
        self.result -= num;
        print("result \(self.result)");
        return self;
    }
    
    func multiply(_ num:Int) -> XZCalculator {
        self.result *= num;
        print("result \(self.result)");
        return self;
    }
    
    func divide(_ num:Int) -> XZCalculator {
        self.result /= num;
        print("result \(self.result)");
        return self;
    }
}
